cli/src/cli.rs

```diff
@@ -218,73 +298,39 @@ pub async fn terraform_apply(workspace: String, _sub: &ArgMatches) -> Result<()>
     Ok(())
 }
 
-async fn info_print(output: TerraformStateValue) -> Result<()> {
-    let mut instance_table = Table::new();
-    instance_table.add_row(row!["Name", "Type", "FlakeAttr", "Private IP", "Public IP"]);
-
-    for (key, val) in output.instances.iter() {
-        instance_table.add_row(row![
-            key,
-            val.instance_type,
-            val.flake_attr,
-            val.private_ip,
-            val.public_ip,
-        ]);
-    }
-
-    instance_table.printstd();
-
-    let mut asg_table = Table::new();
-
-    asg_table.add_row(row![
-        "Id",
-        "Type",
-        "AZ",
-        "State",
-        "Status",
-        "Protected",
-        "PrivateIp",
-        "PublicIp",
-        "asgSuffix"
-    ]);
-
-    for (_key, val) in output.asgs.iter() {
-        let asg_prefix = format!(
-            "client-{}-{}",
-            val.region,
-            val.instance_type.replace(".", "-")
-        );
-        let asg_suffix = _key
-            .strip_prefix(&asg_prefix)
-            .unwrap_or_else(|| "ERROR")
-            .replace("-", "");
-
-        let info = info::asg_info(val.arn.as_str(), val.region.as_str()).await;
-
-        let instance_ids = info.iter().map(|x| x.instance_id.as_str()).collect();
-        let instances = info::instance_info(instance_ids, val.region.as_str()).await;
-
-        for asgi in info {
-            let instance = instances
-                .iter()
-                .find(|x| x.instance_id == Some(asgi.instance_id.clone()))
-                .unwrap();
-
-            asg_table.add_row(row![
-                asgi.instance_id,
-                asgi.instance_type.unwrap_or_default(),
-                asgi.availability_zone,
-                asgi.lifecycle_state,
-                asgi.health_status,
-                asgi.protected_from_scale_in,
-                instance.private_ip_address.as_ref().unwrap_or(&"".to_string()),
-                instance.public_ip_address.as_ref().unwrap_or(&"".to_string()),
-                asg_suffix,
-            ]);
-            // asg_table.add_row(row![key, val.instance_type, val.flake_attr, val.count,]);
+async fn info_print(cluster: ClusterHandle, json: bool) -> Result<()> {
+    if json {
+        let stdout = io::stdout();
+        let handle = stdout.lock();
+        let mut cluster = cluster.await??;
+        let nodes = cluster
+            .nodes
+            .into_iter()
+            .map(|mut node| {
+                node.nomad_client = None;
+                node
+            })
+            .collect();
+        cluster.nodes = nodes;
+        serde_json::to_writer_pretty(handle, &cluster)?;
+    } else {
+        let mut instance_table = Table::new();
+        instance_table.add_row(row!["Name", "Private IP", "Public IP"]);
```

Comment:
Not a fan of getting rid of all this information... i'd like at least the type, region, and suffix to remain. the rest could be in the json output.

Reply:
I had a feeling someone would complain, but I also felt like the previous table was a bit overwhelming, so my strategy was to remove almost everything and wait for someone to complain :sweat_smile: 

What you request sounds reasonable.

Reply:
After some additional reflection, some of the requested fields may not be available when on prem. I suppose I could still implement this and just construct the table fields based on the BitteProvider